### ISSUE TITLE
Unable to migrate from locally hosted git repo (e.g. bitbucket server)

### ISSUE THREAD

**Issue by ollie1** [gogs/gogs#6841]
### Gogs version

0.12.5

### Git version

- Server: 2.32.0
- Client: 2.32.0

### Operating system

Alpine 3.14.3

### Database

sqlite3

### Describe the bug

Firstly, thanks for a fantastic project and for your hard work building and maintaining it!

There is a regression in 0.12.5 caused by the fix to https://github.com/gogs/gogs/issues/6754 where we are no longer able to migrate a repo from a locally hosted git repository - in this case, bitbucket server, but it would equally apply to gitlab, or even another gogs instance.  This is because the fix to the above issue prevents migrating from any url which resolves to a local IP address.

### To reproduce

Attempt to create a repo migrated from a locally hosted git url (in this case using the API, but I would expect the UI to have the same issue).  e.g. using the API endpoint
`POST /api/v1/repos/migrate`
```json
    {
        "clone_addr": "https://bitbucket-server.corp.local/scm/PROJ/my-repo.git",
        "auth_username": "user",
        "auth_password": "pass",
        "uid": 1,
        "repo_name": "my-repo",
        "mirror": true,
        "private": true
      }
```

Instead of the expected behaviour below, an HTTP 422 code is received, with the message `invalid clone address [is_url_error: true, is_invalid_path: false, is_permission_denied: false]`.  This seems to have been caused by the change here: https://github.com/gogs/gogs/pull/6812/files#diff-32fb9b682e2e195664245ddf478b01b5a9916614f5226eed22ae5b9786e4ed4dR74 to fix a security issue. Without details of what the security issue is, it's hard to know if this is an expected side effect of the fix, but it seems like a big loss of functionality which worked perfectly fine in 0.12.4.

### Expected behavior

The repo is created successfully and contains the content from the locally hosted git repo.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by unknwon** [comment]
The fix of this regression will be included in 0.12.6, currently plan to happen on March 19-20th (not a commitment).

**Comment by unknwon** [comment]
The 0.12.6 has been released that includes the patch of the reported issue.
